### PR TITLE
MODELIX-584 Update roadmap

### PR DESCRIPTION
**Pull request by nkoester**

### CODE REVIEW

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 1]
```diff
@@ -1,98 +1,102 @@
...
+
+// ------------- 24.1
+[#modelql,reftext="ModelQL"]
+=== ModelQL (estimated 24.1)
+
+With the modelix 23.2 release we https://modelix.org/blog/2023/11/02/modelix-platform-release-23.2/#modelql[introduced ModelQL]:
+A dynamic way of loading parts of the model on demand while reducing the number of requests to a minimum.
+To learn more, head over to the https://docs.modelix.org/modelix/main/core/explanation/modelql.html[ModelQL documentation] and the corresponding https://docs.modelix.org/modelix/main/core/howto/modelql.html[How-Tos].
+While the early version allowed us to provide a quick win for large models, we were not able to finish this feature fully.
+Especially https://docs.modelix.org/modelix/main/core/howto/modelql-writing.html[writing support] currently needs some further work.
```
Comment (mhuster23) [inline docs/modules/ROOT/pages/main/roadmap.adoc]: I think the writing support is pretty much done for now.
The functionality of modelql is there, but we should work on stuff like the debugging experience.

File: docs/modules/ROOT/pages/main/development.adoc [hunk 2]
```diff
@@ -2,28 +2,30 @@
 
 == Development
 
-The modelix project is developed open-source under the https://www.apache.org/licenses/LICENSE-2.0.html[Apache-2.0 license].
-All components and software of the platform as well as issue tracking and project management are done in repositories in our https://github.com/modelix[GitHub organisation].
+The modelix project is developed open-source under the https://www.apache.org/licenses/LICENSE-2.0.html[Apache-2.0 license^].
```
Comment (odzhychko) [inline docs/modules/ROOT/pages/main/development.adoc:5]: nice to know the [`^` blank window shorthand](https://docs.asciidoctor.org/asciidoc/latest/macros/link-macro-attribute-parsing/#blank-window-shorthand).

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 3]
```diff
@@ -1,98 +1,102 @@
...
+* *MPS 2023 / 2024 Support*: Provide support in modelix for newer MPS versions
+* *LIonWeb* (estimated 24.1): Integration with other web components via a new LionWeb bulk API adapter
+// ------------- 24.2
+* *Model-server services*: Streamlined interfaces and structure for services working on model data, providing for example model analysis, checking, and transformation
+* *Meta-model in model-server*: A way to store the meta-model of corresponding model content in the model-server (foundation for Language Evolution).
+// ------------- 25.1
+* *GraphQL interface*: An endpoint for smaller web applications, leveraging the available tooling ecosystem
+* *Projectional text editors in browser*: Projectional text editors for browsers, using Kotlin multiplatform to edit models directly or through server-side rendering
+// ------------- 25.2
+* *Graphical Editors*: We plan to provide the data synchronization and the generation of a TypeScript model API to enable graphical editors on the wev.
```
Comment (odzhychko) [inline docs/modules/ROOT/pages/main/roadmap.adoc]:
```suggestion
* *Graphical Editors*: We plan to provide the data synchronization and the generation of a TypeScript model API to enable graphical editors on the web.
```

File: docs/modules/ROOT/pages/main/roadmap.adoc (hunk 1 above)
Comment (odzhychko) [reply docs/modules/ROOT/pages/main/roadmap.adoc]: You could also say we are working on making it more mature/stabalizing it (aka. bugfixing and peformance)

File: docs/modules/ROOT/pages/main/roadmap.adoc (hunk 3 above)
Comment (benedekh) [inline docs/modules/ROOT/pages/main/roadmap.adoc]:
```suggestion
* *Graphical Editors*: We plan to provide the data synchronization and the generation of a TypeScript model API to enable graphical editors on the web.
```

File: docs/modules/ROOT/pages/main/development.adoc [hunk 4]
```diff
@@ -2,28 +2,30 @@
...
-Generally, we follow the https://en.wikipedia.org/wiki/Software_versioning[_Semantic versioning_] scheme for releases.
+Generally, we follow the https://en.wikipedia.org/wiki/Software_versioning[_Semantic versioning_^] scheme for releases.
 However, anything based on JetBrains MPS (e.g. MPS plugins or extensions) are prefixed with the corresponding MPS major and minor version number followed by our own patch version number.
-For example, the https://github.com/modelix/modelix/packages/1077382[`org.modelix.mps-model-plugin`] version `2021.2.103` is the `103rd` release which is compatible with MPS `2021.2.*`.
-
+For example, the https://github.com/modelix/modelix/packages/1077382[`org.modelix.mps-model-plugin`^] version `2021.2.103` is the `103rd` release which is compatible with MPS `2021.2.*`.
 
 To make it easier for developers, the modelix platform is currently released twice a year as a single "platform release" which references compatible component versions.
-We aim to provide a Spring release `R1` and an Autumn release `R2`.
+We aim to provide a Spring release (e.g. `24.1`) and an Autumn release (e.g. `24.2`).
```
Comment (benedekh) [inline docs/modules/ROOT/pages/main/development.adoc:21]:
```suggestion
We aim to provide a Spring release (i.e. `24.1`) and an Autumn release (i.e. `24.2`).
```

i.e. = that is
e.g. = for example

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 5]
```diff
@@ -1,98 +1,102 @@
...
+While the early version allowed us to provide a quick win for large models, we were not able to finish this feature fully.
+Especially https://docs.modelix.org/modelix/main/core/howto/modelql-writing.html[writing support] currently needs some further work.
+We also want to integrate ModelQL in our https://github.com/modelix/modelix-samples[samples project] to present a graspable example beyond the documentation.
+
+
+=== MPS 2023 / 2024 Support (estimated 24.1)
+
+In a larger effort to become MPS version independent, we prepare our plugins and interfaces with MPS to provide preliminary support for MPS 2023 for the modelix 24.1 release.
+This effort is needed to provide adequate support today, but in the following release 24.2 we plan to fully transition away from dedicated MPS plugins over to https://plugins.jetbrains.com/docs/intellij/intellij-platform.html?from=IJPluginTemplate[IntelliJ Platform] plugins.
+IntelliJ Platform plugins do not require to be written in MPS and allow to be created using Kotlin - just like everything else in modelix.
```
Comment (benedekh) [inline docs/modules/ROOT/pages/main/roadmap.adoc]:
```suggestion
IntelliJ Platform plugins do not require to be written in MPS and are allowed to be created using Kotlin - just like everything else in modelix.
```

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 6]
```diff
@@ -1,98 +1,102 @@
...
+Consequently, for smaller web applications we consider to provide a simple https://graphql.org/[GraphQL] endpoint, which can provide immediate benefit.
+Especially the access to a powerful tooling ecosystem with GUIs, editor integrations, code generation, linting, analytics, and more is a direct win.
+
+However, it is important to note, that this approach will not scale:
+As soon as the model size grows or the strong integration requirements exist, other tools such as our <<modelql>> or the <<services>> are needed.
+
+=== Projectional text editors in browser (estimated 25.1)
+Similar to our previously developed https://github.com/modelix/modelix/tree/mps/2020.3/mps/org.modelix.notation[notation language] prototype (based on the old shadow models), we want to provide
+A default notation is used for concepts without a web specific notation, just as MPS does it.
+We envision projectional text editor components implemented in Kotlin multiplatform for editing models in the browser, either directly or via server side rendering.
```
Comment (benedekh) [inline docs/modules/ROOT/pages/main/roadmap.adoc]:
```suggestion
We envision projectional text editor components implemented in Kotlin multiplatform for editing models in the browser, either directly or via server-side rendering.
```

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 7]
```diff
@@ -1,98 +1,102 @@
...
+IntelliJ Platform plugins do not require to be written in MPS and allow to be created using Kotlin - just like everything else in modelix.
+All MPS interfacing parts will use the comparably stable APIs provided by MPS to query and modify MPS models, such as the https://www.jetbrains.com/help/mps/open-api-accessing-models-from-code.html:[Open API] and the https://www.jetbrains.com/help/mps/smodel-language.html:[SModel].
+As a result, we do not have to migrate and maintain MPS plugins for multiple MPS versions at the same time.
+Instead, we have a single Kotlin codebase which can provide the same functionality to all MPS versions.
+This effort will reduce complexity and potential for bugs and allow us to better test and validate our MPS interfacing plugins.
+
+
+=== LIonWeb (estimated 24.1)
+Modelix focuses on a good integration with MPS.
+But if you are interested in the integration of other modeling technologies then check out the LIonWeb project (short for Language Interfaces on the web).
```
Comment (odzhychko) [inline docs/modules/ROOT/pages/main/roadmap.adoc]: A link to the LIonWeb project could be nice.

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 8]
```diff
@@ -1,98 +1,102 @@
...
+=== GraphQL interface (estimated 25.1)
+As with all other parts of modelix, we prefer using established and state-of-the-art tools.
+Consequently, for smaller web applications we consider to provide a simple https://graphql.org/[GraphQL] endpoint, which can provide immediate benefit.
+Especially the access to a powerful tooling ecosystem with GUIs, editor integrations, code generation, linting, analytics, and more is a direct win.
+
+However, it is important to note, that this approach will not scale:
+As soon as the model size grows or the strong integration requirements exist, other tools such as our <<modelql>> or the <<services>> are needed.
+
+=== Projectional text editors in browser (estimated 25.1)
+Similar to our previously developed https://github.com/modelix/modelix/tree/mps/2020.3/mps/org.modelix.notation[notation language] prototype (based on the old shadow models), we want to provide
```
Comment (odzhychko) [inline docs/modules/ROOT/pages/main/roadmap.adoc]: > Similar to our previously developed notation language prototype, we want to provide

The sentence is incomplete.

File: docs/modules/ROOT/pages/main/development.adoc (hunk 2 above)
Comment (languitar) [reply docs/modules/ROOT/pages/main/development.adoc:5]: I hate it when pages try to decide for me whether I want a new tab or not. I can easily select that on my own with either clicking the LMB or the wheel

File: docs/modules/ROOT/images/2024_01_roadmap.png
Comment (languitar) [inline docs/modules/ROOT/images/2024_01_roadmap.png:1]: Should probably be "MPS-independent"

File: docs/modules/ROOT/pages/main/development.adoc (hunk 4 above)
Comment (languitar) [reply docs/modules/ROOT/pages/main/development.adoc:21]: Probably depends on whether the sentence tries to show the general pattern or the next year.

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 9]
```diff
@@ -1,98 +1,102 @@
...
-The experience from MPS shows that the requirements for a graphical editor are too diverse to support all of them with a single framework. There are many different frameworks available in the web world where you can choose the one that fits your use case.
+In the following we shortly list these key priorities.
+Below you can find more detailed explanations for each of these.
+
+// ------------- 24.1
+* *ModelQL*: Dynamic query language for on-demand loading overcoming issues associated with upfront data retrieval and optimizing performance
+* *MPS 2023 / 2024 Support*: Provide support in modelix for newer MPS versions
+* *LIonWeb* (estimated 24.1): Integration with other web components via a new LionWeb bulk API adapter
+// ------------- 24.2
+* *Model-server services*: Streamlined interfaces and structure for services working on model data, providing for example model analysis, checking, and transformation
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/roadmap.adoc]: "example model analysis" is a bit imprecise. Also, the word "service" should probably be defined.

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 10]
```diff
@@ -1,98 +1,102 @@
...
+* *Meta-model in model-server*: A way to store the meta-model of corresponding model content in the model-server (foundation for Language Evolution).
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/roadmap.adoc]:
```suggestion
* *Meta-model in model-server*: A way to store the meta-model of corresponding model content in the model-server (foundation for language evolution).
```

and also for other things? So maybe add "e.g." or something similar.

File: docs/modules/ROOT/pages/main/roadmap.adoc [hunk 11]
```diff
@@ -1,98 +1,102 @@
...
+// ------------- 24.2
+[#services, reftext="model-server services"]
+=== Model-server services (estimated 24.2)
+Modelix already allows you to access model data from any web application, either untyped (i.e. on the M3 layer), or typed (i.e. on the M2 layer or domain layer).
+As soon as you have model content in a web application the question regarding correct scoping, typing, and other checks arises.
+In MPS we grew accustomed to easy integration of these different language aspects.
+In modelix we envision `model-server` *services* to do this type of model analysis, checking, and eventually transformation.
+In fact, this can already be done today:
+Any application accessing model data from a `model-server` can provide external APIs for the above-mentioned features.
+However, we plan to streamline interfaces and approaches used by efforts that typical services can provide.
```
Comment (languitar) [inline docs/modules/ROOT/pages/main/roadmap.adoc]: "efforts" is a bit wordy

File: docs/modules/ROOT/pages/main/development.adoc (hunk 2 above)
Comment (odzhychko) [reply docs/modules/ROOT/pages/main/development.adoc:5]: > Configuring a link that points to a location outside the current site is common practice to avoid disrupting the reader’s flow.

(information comes from same page as above)

Seems to be a common practice...

Comment (languitar) [reply docs/modules/ROOT/pages/main/development.adoc:5]: I've seen that. But still, sometime I just want to follow in the same tab and that things makes this impossible to me. If not specified, I could freely decide.

File: docs/modules/ROOT/pages/main/development.adoc (hunk 4 above)
Comment (nkoester) [reply docs/modules/ROOT/pages/main/development.adoc:21]: yes, goal was the general pattern

Comment (benedekh) [reply docs/modules/ROOT/pages/main/development.adoc:21]: Fine for me :)

**Review by benedekh — APPROVED** [review]

**Review by odzhychko — APPROVED** [review]
